### ISSUE TITLE
9$: warn on 'keyof T[]' in type

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#1218]
In a TypeScript type, `keyof T[]` is the same as `number` (I think). The user likely meant `(keyof T)[]`. We should emit a warning, telling them to write either `number` or add parentheses.

Inspired by this article: https://www.totaltypescript.com/array-types-in-typescript
> Using keyof with T[] can lead to unexpected results.
> ```typescript
> const result: keyof Person[] = ["id", "name"];
> ```

**Comment by Jetoson** [comment]
@strager May I work on this?
